schema

```diff
@@ -0,0 +1,357 @@
+    },
+    "data": {
+      "description": "The document's \"primary data\" is a representation of the resource or collection of resources targeted by a request.",
+      "oneOf": [
+        {
+          "$ref": "#/definitions/resource"
+        },
+        {
+          "description": "An array of resource objects, an array of resource identifier objects, or an empty array ([]), for requests that target resource collections.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/resource"
+          },
+          "uniqueItems": true
+        }
+      ]
+    },
+    "resource": {
+      "description": "\"Resource objects\" appear in a JSON API document to represent resources.",
+      "type": "object",
+      "required": [
+        "type",
+        "id"
+      ],
+      "properties": {
+        "type": {
+          "type": "string"
+        },
+        "id": {
+          "type": "string"
+        },
+        "attributes": {
+          "$ref": "#/definitions/attributes"
+        },
+        "relationships": {
+          "$ref": "#/definitions/relationships"
+        },
+        "links": {
+          "$ref": "#/definitions/links"
+        },
+        "meta": {
+          "$ref": "#/definitions/meta"
+        }
+      },
+      "additionalProperties": false
+    },
+		
+    "links": {
+      "description": "A resource object **MAY** contain references to other resource objects (\"relationships\"). Relationships may be to-one or to-many. Relationships can be specified by including a member in a resource's links object.",
+      "type": "object",
+      "properties": {
+        "self": {
+          "description": "A `self` member, whose value is a URL for the relationship itself (a \"relationship URL\"). This URL allows the client to directly manipulate the relationship. For example, it would allow a client to remove an `author` from an `article` without deleting the people resource itself.",
+          "type": "string",
+          "format": "uri"
+        },
+        "related": {
+          "$ref": "#/definitions/link"
+        }
+      },
+      "additionalProperties": true
+    },
+		"link": {
+			"description": "A link **MUST** be represented as either: a string containing the link's URL or a link object.",
+			"oneOf": [
+				{
+					"description": "A string containing the link's URL.",
+					"type": "string",
+					"format": "uri"
+				},
+				{
+					"type": "object",
+					"required": [
+						"href"
+					],
+					"properties": {
+						"href": {
+							"description": "A string containing the link's URL.",
+							"type": "string",
+							"format": "uri"
+						},
+						"meta": {
+							"$ref": "#/definitions/meta"
+						}
+					}
+				}
+			]
+		},
+
+    "attributes": {
+			"description": "Members of the attributes object (\"attributes\") represent information about the resource object in which it's defined.",
+      "type": "object",
+			"patternProperties": {
+				"^\\w[\\w-_]*$": {
```

Comment:
Is it too strict to filter out the [reserved member names](http://jsonapi.org/format/#document-resource-object-attributes) here? That is adding `relationships` and `links` to the regex.